lib/portlayer/exec/container.go

```diff
@@ -220,3 +221,27 @@ func (c *Container) Stop(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *Container) VmRemove(ctx context.Context) error {
+	containersLock.Lock()
+	c.Lock()
+	defer containersLock.Lock()
```

Comment:
defer occurs like a stack as far as I know. Which means that the unlocks will occur in the correct order. Unfortunately upong reading this I realized that that defer is actually calling a lock... fixing this now. I just had a conversation about how to move the locks around with hasan so this is changing anyways.